### PR TITLE
[qtdeclarative] add long path warning

### LINKED ISSUE #29454: [ qtdeclarative[core]] Cannot open include file: 'qquicksaturationlightnesspicker_p.h': No such file or directory
**Host Environment**

- OS: Windows11
- Compiler: msvc vc413

**To Reproduce**

`.\vcpkg install pcl[apps,cuda,opengl,openni2,pcap,qt,simulation,tools,visualization,vtk]:x64-windows ` or
`.\vcpkg install qt:x64-windows `

**Failure logs**

-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: D:/ProgramFiles/CMake/bin/cmake.exe --build . --config Debug --target install -- -v -j13
    Working Directory: D:/ProgramFiles/vcpkg/buildtrees/qtdeclarative/x64-windows-dbg
    See logs for more information:
      D:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  installed/x64-windows/share/qtbase/qt_install_submodule.cmake:337 (vcpkg_cmake_install)
  ports/qtdeclarative/portfile.cmake:22 (qt_install_submodule)
  scripts/ports.cmake:147 (include)
**Additional context**
[2843/2933] D:\ProgramFiles\Microsoft\VisualStudio\2022\Community\VC\Tools\MSVC\14.34.31933\bin\Hostx64\x64\cl.exe   /TP -DQT_ASCII_CAST_WARNINGS -DQT_BUILDING_QT -DQT_BUILD_QUICKDIALOGS2QUICKIMPL_LIB -DQT_CORE_LIB -DQT_DEPRECATED_WARNINGS -DQT_DEPRECATED_WARNINGS_SINCE=0x070000 -DQT_DISABLE_DEPRECATED_BEFORE=0x040800 -DQT_GUI_LIB -DQT_LEAN_HEADERS=1 -DQT_MOC_COMPAT -DQT_NETWORK_LIB -DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_TO_ASCII -DQT_NO_EXCEPTIONS -DQT_NO_JAVA_STYLE_ITERATORS -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_OPENGL_LIB -DQT_QMLINTEGRATION_LIB -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICKCONTROLS2IMPL_LIB -DQT_QUICKDIALOGS2UTILS_LIB -DQT_QUICKTEMPLATES2_LIB -DQT_QUICK_LIB -DQT_USE_QSTRINGBUILDER -DQuickDialogs2QuickImpl_EXPORTS -DUNICODE -DWIN32 -DWIN64 -D_CRT_SECURE_NO_WARNINGS -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_USE_MATH_DEFINES -D_WIN64 -D_WINDLL -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\quickdialogs2\quickdialogs2quickimpl\QuickDialogs2QuickImpl_autogen\include -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickDialogs2QuickImpl -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\src\here-src-6-a31aea0fcd.clean\src\quickdialogs2\quickdialogs2quickimpl -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\quickdialogs2\quickdialogs2quickimpl -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickDialogs2QuickImpl\6.4.2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickDialogs2QuickImpl\6.4.2\QtQuickDialogs2QuickImpl -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\qml -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQml\6.4.2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQml\6.4.2\QtQml -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQml -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQmlIntegration -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\src\here-src-6-a31aea0fcd.clean\src\qmlintegration -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\qmlintegration -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuick -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQmlModels -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\quick -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuick\6.4.2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuick\6.4.2\QtQuick -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\qmlmodels -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQmlModels\6.4.2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQmlModels\6.4.2\QtQmlModels -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickTemplates2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\quicktemplates2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickTemplates2\6.4.2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickTemplates2\6.4.2\QtQuickTemplates2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\quickcontrols2impl -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickControls2Impl\6.4.2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickControls2Impl\6.4.2\QtQuickControls2Impl -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickControls2Impl -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickDialogs2Utils -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\quickdialogs2\quickdialogs2utils -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickDialogs2Utils\6.4.2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickDialogs2Utils\6.4.2\QtQuickDialogs2Utils -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtCore\6.4.2 -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtCore\6.4.2\QtCore -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtCore -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6 -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\share\Qt6\mkspecs\win32-msvc -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtNetwork -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtGui -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtOpenGL -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtGui\6.4.2 -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtGui\6.4.2\QtGui -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -MDd /W3 /EHs-c- /wd4530 /wd4577 -Zc:__cplusplus -permissive- -utf-8 -FS -Zc:rvalueCast -Zc:inline -Zc:strictStrings -Zc:throwingNew -Zc:referenceBinding -Zc:externConstexpr -Zc:wchar_t -bigobj -std:c++17 /bigobj /YuD:/ProgramFiles/vcpkg/buildtrees/qtdeclarative/x64-windows-dbg/src/quickdialogs2/quickdialogs2quickimpl/CMakeFiles/QuickDialogs2QuickImpl.dir/cmake_pch.hxx /FpD:/ProgramFiles/vcpkg/buildtrees/qtdeclarative/x64-windows-dbg/src/quickdialogs2/quickdialogs2quickimpl/CMakeFiles/QuickDialogs2QuickImpl.dir/./cmake_pch.cxx.pch /FID:/ProgramFiles/vcpkg/buildtrees/qtdeclarative/x64-windows-dbg/src/quickdialogs2/quickdialogs2quickimpl/CMakeFiles/QuickDialogs2QuickImpl.dir/cmake_pch.hxx /showIncludes /Fosrc\quickdialogs2\quickdialogs2quickimpl\CMakeFiles\QuickDialogs2QuickImpl.dir\quickdialogs2quickimpl_qmltyperegistrations.cpp.obj /Fdsrc\quickdialogs2\quickdialogs2quickimpl\CMakeFiles\QuickDialogs2QuickImpl.dir\ /FS -c D:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\quickdialogs2\quickdialogs2quickimpl\quickdialogs2quickimpl_qmltyperegistrations.cpp
FAILED: src/quickdialogs2/quickdialogs2quickimpl/CMakeFiles/QuickDialogs2QuickImpl.dir/quickdialogs2quickimpl_qmltyperegistrations.cpp.obj 
D:\ProgramFiles\Microsoft\VisualStudio\2022\Community\VC\Tools\MSVC\14.34.31933\bin\Hostx64\x64\cl.exe   /TP -DQT_ASCII_CAST_WARNINGS -DQT_BUILDING_QT -DQT_BUILD_QUICKDIALOGS2QUICKIMPL_LIB -DQT_CORE_LIB -DQT_DEPRECATED_WARNINGS -DQT_DEPRECATED_WARNINGS_SINCE=0x070000 -DQT_DISABLE_DEPRECATED_BEFORE=0x040800 -DQT_GUI_LIB -DQT_LEAN_HEADERS=1 -DQT_MOC_COMPAT -DQT_NETWORK_LIB -DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_TO_ASCII -DQT_NO_EXCEPTIONS -DQT_NO_JAVA_STYLE_ITERATORS -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_OPENGL_LIB -DQT_QMLINTEGRATION_LIB -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICKCONTROLS2IMPL_LIB -DQT_QUICKDIALOGS2UTILS_LIB -DQT_QUICKTEMPLATES2_LIB -DQT_QUICK_LIB -DQT_USE_QSTRINGBUILDER -DQuickDialogs2QuickImpl_EXPORTS -DUNICODE -DWIN32 -DWIN64 -D_CRT_SECURE_NO_WARNINGS -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_USE_MATH_DEFINES -D_WIN64 -D_WINDLL -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\quickdialogs2\quickdialogs2quickimpl\QuickDialogs2QuickImpl_autogen\include -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickDialogs2QuickImpl -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\src\here-src-6-a31aea0fcd.clean\src\quickdialogs2\quickdialogs2quickimpl -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\quickdialogs2\quickdialogs2quickimpl -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickDialogs2QuickImpl\6.4.2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickDialogs2QuickImpl\6.4.2\QtQuickDialogs2QuickImpl -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\qml -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQml\6.4.2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQml\6.4.2\QtQml -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQml -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQmlIntegration -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\src\here-src-6-a31aea0fcd.clean\src\qmlintegration -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\qmlintegration -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuick -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQmlModels -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\quick -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuick\6.4.2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuick\6.4.2\QtQuick -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\qmlmodels -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQmlModels\6.4.2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQmlModels\6.4.2\QtQmlModels -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickTemplates2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\quicktemplates2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickTemplates2\6.4.2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickTemplates2\6.4.2\QtQuickTemplates2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\quickcontrols2impl -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickControls2Impl\6.4.2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickControls2Impl\6.4.2\QtQuickControls2Impl -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickControls2Impl -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickDialogs2Utils -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\quickdialogs2\quickdialogs2utils -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickDialogs2Utils\6.4.2 -ID:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickDialogs2Utils\6.4.2\QtQuickDialogs2Utils -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtCore\6.4.2 -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtCore\6.4.2\QtCore -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtCore -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6 -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\share\Qt6\mkspecs\win32-msvc -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtNetwork -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtGui -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtOpenGL -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtGui\6.4.2 -external:ID:\ProgramFiles\vcpkg\installed\x64-windows\include\Qt6\QtGui\6.4.2\QtGui -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -MDd /W3 /EHs-c- /wd4530 /wd4577 -Zc:__cplusplus -permissive- -utf-8 -FS -Zc:rvalueCast -Zc:inline -Zc:strictStrings -Zc:throwingNew -Zc:referenceBinding -Zc:externConstexpr -Zc:wchar_t -bigobj -std:c++17 /bigobj /YuD:/ProgramFiles/vcpkg/buildtrees/qtdeclarative/x64-windows-dbg/src/quickdialogs2/quickdialogs2quickimpl/CMakeFiles/QuickDialogs2QuickImpl.dir/cmake_pch.hxx /FpD:/ProgramFiles/vcpkg/buildtrees/qtdeclarative/x64-windows-dbg/src/quickdialogs2/quickdialogs2quickimpl/CMakeFiles/QuickDialogs2QuickImpl.dir/./cmake_pch.cxx.pch /FID:/ProgramFiles/vcpkg/buildtrees/qtdeclarative/x64-windows-dbg/src/quickdialogs2/quickdialogs2quickimpl/CMakeFiles/QuickDialogs2QuickImpl.dir/cmake_pch.hxx /showIncludes /Fosrc\quickdialogs2\quickdialogs2quickimpl\CMakeFiles\QuickDialogs2QuickImpl.dir\quickdialogs2quickimpl_qmltyperegistrations.cpp.obj /Fdsrc\quickdialogs2\quickdialogs2quickimpl\CMakeFiles\QuickDialogs2QuickImpl.dir\ /FS -c D:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\src\quickdialogs2\quickdialogs2quickimpl\quickdialogs2quickimpl_qmltyperegistrations.cpp
cl : Command line warning D9025 : overriding '/EHs' with '/EHs-'
cl : Command line warning D9025 : overriding '/EHc' with '/EHc-'
D:\ProgramFiles\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickDialogs2QuickImpl\6.4.2\QtQuickDialogs2QuickImpl\private/qquicksaturationlightnesspicker_p.h(1): fatal error C1083: Cannot open include file: '../../../../../../src/here-src-6-a31aea0fcd.clean/src/quickdialogs2/quickdialogs2quickimpl/qquicksaturationlightnesspicker_p.h': No such file or directory

### PR DESCRIPTION
**Pull request by Adela0814**
Fixes https://github.com/microsoft/vcpkg/issues/29454

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/qtdeclarative/portfile.cmake [hunk 1]
```diff
@@ -1,6 +1,13 @@
 set(SCRIPT_PATH "${CURRENT_INSTALLED_DIR}/share/qtbase")
 include("${SCRIPT_PATH}/qt_install_submodule.cmake")
 
+string(LENGTH "${CURRENT_BUILDTREES_DIR}" buildtrees_path_length)
+if(buildtrees_path_length GREATER 44 AND CMAKE_HOST_WIN32)
+    vcpkg_buildpath_length_warning(44)
+    message(WARNING "The ${PORT} source was will be extracted to ${CURRENT_BUILDTREES_DIR} , which has more then 44 characters in length.")
+    message(FATAL_ERROR "terminating due to ${CURRENT_BUILDTREES_DIR} being too long.")
```
Comment (Neumann-A) [inline ports/qtdeclarative/portfile.cmake:8]: Not helpful. Requires at least a hint to `--x-buildtree-root=<shortpath>`

File: ports/qtdeclarative/portfile.cmake [hunk 2]
```diff
@@ -1,6 +1,13 @@
 set(SCRIPT_PATH "${CURRENT_INSTALLED_DIR}/share/qtbase")
 include("${SCRIPT_PATH}/qt_install_submodule.cmake")
 
+string(LENGTH "${CURRENT_BUILDTREES_DIR}" buildtrees_path_length)
+if(buildtrees_path_length GREATER 44 AND CMAKE_HOST_WIN32)
+    vcpkg_buildpath_length_warning(44)
```
Comment (Neumann-A) [inline ports/qtdeclarative/portfile.cmake:6]: Did you try it out until it failed or where does the number come from?

Comment (Neumann-A) [inline ports/qtdeclarative/portfile.cmake:6]:
```suggestion
```
doesn't seem necessary considering the rest.

Comment (Adela0814) [reply ports/qtdeclarative/portfile.cmake:6]: This is the path length I tested until failure.

**Comment by autoantwort** [comment]
Why not using `vcpkg_buildpath_length_warning`?

**Comment by Adela0814** [comment]
> Why not using `vcpkg_buildpath_length_warning`?

The content of this function is already included in the rest.

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by Neumann-A** [comment]
> Why not using vcpkg_buildpath_length_warning?

Also we want a hard error instead of a warning. Maybe vcpkg should provide `vcpkg_check_buildpath_length` instead with a possibility to select the error/warning mode.

**Comment by JonLiu1993** [comment]
````
Building qtdeclarative[core]:x64-windows...
CMake Warning at ports/qtdeclarative/portfile.cmake:6 (message):
  The qtdeclarative source was will be extracted to
  F:/Feature-test/at/vcpkg/buildtrees/qtdeclarative , which has more then 44
  characters in length.
Call Stack (most recent call first):
  scripts/ports.cmake:147 (include)
````

**Comment by JavierMatosD** [comment]
Adding team review to consider Neumann-A suggestion.

**Comment by JavierMatosD** [comment]
Please update the PR to use the ```vcpkg_buildpath_length_warning```.

**Comment by Adela0814** [comment]
> Please update the PR to use the `vcpkg_buildpath_length_warning`.

Will the function(`vcpkg_buildpath_length_warning`) be updated? If not, can I keep the current changes, just add a line to call this function in my changes?

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JavierMatosD** [comment]
> > Please update the PR to use the `vcpkg_buildpath_length_warning`.
> 
> @JavierMatosD Will the function(`vcpkg_buildpath_length_warning`) be updated? If not, can I keep the current changes, just add a line to call this function in my changes?

 `vcpkg_buildpath_length_warning` already emits a warning for the user. We don't want the hard error at the moment.

File: ports/qtdeclarative/portfile.cmake [hunk 3]
```diff
@@ -1,6 +1,13 @@
...
+    message(WARNING "The ${PORT} source was will be extracted to ${CURRENT_BUILDTREES_DIR} , which has more then 44 characters in length.")
+    message(FATAL_ERROR "terminating due to ${CURRENT_BUILDTREES_DIR} being too long. Pass --x-buildtrees-root=<shortpath> to vcpkg")
+endif()
```
Comment (JavierMatosD) [inline ports/qtdeclarative/portfile.cmake:9]:
```suggestion
vcpkg_buildpath_length_warning(44)
```

**Comment by Neumann-A** [comment]
> vcpkg_buildpath_length_warning already emits a warning for the user. We don't want the hard error at the moment.

So you just want the user to build the port for 5 mins and than have the build fail with a strange error instead of aborting early?

File: ports/qtdeclarative/portfile.cmake (hunk 3 above)
Comment (JavierMatosD) [reply ports/qtdeclarative/portfile.cmake:9]: We could add something to vcpkg.json so the tool can bail early?

**Comment by JavierMatosD** [comment]
> > vcpkg_buildpath_length_warning already emits a warning for the user. We don't want the hard error at the moment.
> 
> So you just want the user to build the port for 5 mins and than have the build fail with a strange error instead of aborting early?



> > vcpkg_buildpath_length_warning already emits a warning for the user. We don't want the hard error at the moment.
> 
> So you just want the user to build the port for 5 mins and than have the build fail with a strange error instead of aborting early?

To be fair, the message does give instructions to resolve the issue. That said, we could add something to the vcpkg.json so that it can check it before it builds anything.

**Comment by JavierMatosD** [comment]
> > vcpkg_buildpath_length_warning already emits a warning for the user. We don't want the hard error at the moment.
> 
> So you just want the user to build the port for 5 mins and than have the build fail with a strange error instead of aborting early?

To be fair, the message provides instructions to resolve the issue. That said, we could add something to vcpkg.json so it can check before building anything.

**Comment by dg0yt** [comment]
> That said, we could add something to vcpkg.json so it can check before building anything.

Note that this error depends on the platform.

Generally it would be appropriate to allow passing known message type keywords into such functions (`WARNING`,`FATAL_ERROR`).
But the location where the error is raised also interacts with `--only-downloads` mode (until downloads are moved to manifest...).

**Pull request merged**